### Model Validation Technique
There are 3 Technique

1.Train & Test Split

2.K-fold Cross Validation

3.Leave One Out Cross Validation

#### Using a Logistic Model 

#### 1.Evaluate using a train and a test set

In [1]:
# Evaluate using a train and a test set

#Importing the Required Libraries
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

import warnings
warnings.filterwarnings("ignore")  #--to ignore warnings

#loading the Dataset
names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']
Dataset = read_csv('C:/Users/Akaash/Downloads/pima-indians-diabetes_data.csv', names=names)

#Separating the X & Y Variables
array = Dataset.values
X = array[:,0:8]
Y = array[:,8]

#Splitting the Dataset into Train & Test 
test_size = 0.33
seed = 7
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=test_size,random_state=seed)

#Training the Model
model = LogisticRegression()
model.fit(X_train, Y_train)

#Predicting & Getting the Model Score
result = model.score(X_test, Y_test)

#Calculating the Accuracy of the Model
result*100.0

78.74015748031496

Inference: With Train & Test Split The Accuarcy has come 78.74%

#### 2.Evaluate using K-Fold Cross Validation

In [2]:
# Evaluate using Cross Validation

#Importing the Required Libraries
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

#loading the Dataset
names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']
Dataset = read_csv('C:/Users/Akaash/Downloads/pima-indians-diabetes_data.csv', names=names)

#Separating the X & Y Variables
array = Dataset.values
X = array[:,0:8]
Y = array[:,8]

# Creating K-Fold Instance
num_folds = 10
seed = 7
kfold = KFold(n_splits=num_folds, random_state=seed,shuffle=True)

#Training the Model 
model = LogisticRegression(max_iter=400)

#Predicting & Getting the Model Score with kfold
results = cross_val_score(model, X, Y, cv=kfold)


In [3]:
# Accuracy for Each K-Fold
results

array([0.83116883, 0.74025974, 0.74025974, 0.80519481, 0.79220779,
       0.77922078, 0.66233766, 0.80519481, 0.82894737, 0.73684211])

In [4]:
#Average of All k-fold Accuracy to get Final model Accuary
results.mean()*100.0

77.21633629528367

Inference: The K-fold gave the final Accuracy of 77.21%

#### 3.Evaluate using Leave One Out Cross Validation

In [5]:
# Evaluate using Leave One Out Cross Validation

#Importing the Required Libraries
from pandas import read_csv
from sklearn.model_selection import LeaveOneOut
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

#loading the Dataset
names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']
Dataset = read_csv('C:/Users/Akaash/Downloads/pima-indians-diabetes_data.csv', names=names)

#Separating the X & Y Variables
array = Dataset.values
X = array[:,0:8]
Y = array[:,8]

# Creating K-Fold Instance
loocv = LeaveOneOut()

#Training the Model
model = LogisticRegression(max_iter=300)

#Predicting & Getting the Model Score with Leave One Out Cross Validation
results = cross_val_score(model, X, Y, cv=loocv)

In [6]:
#Average of All LooCV Accuracy to get Final model Accuary
results.mean()*100.0

77.60416666666666

Inference:With Leave One Out Cross Validation The Accuarcy has come 77.60%